cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1098,6 +1098,19 @@ def right_scheduler_schema(self, data, **kwargs):
                     f"Scheduling > *Queues section is not appropriate to the Scheduler: {configured_scheduler}."
                 )
 
+    @validates_schema
+    def same_subnet_in_different_queues(self, data, **kwargs):
+        """Validate subnet_ids configured in different queues are the same."""
+        for queues in ["slurm_queues", "aws_batch_queues"]:
+            if data.get(queues, None):
+                queues_list = data.get(queues)
+                subnet_ids_list = []
+                for queue in queues_list:
```

Comment:
could use `for queue in data[queues]`: here and avoid the extra variable created in line 1106

Reply:
Good idea! Changed